x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/analytics/action/PostAnalyticsEventAction.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.application.analytics.action;
+
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.ActionResponse;
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.xcontent.ParseField;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.Objects;
+
+public class PostAnalyticsEventAction extends ActionType<PostAnalyticsEventAction.Response> {
+
+    public static final PostAnalyticsEventAction INSTANCE = new PostAnalyticsEventAction();
+
+    public static final String NAME = "cluster:admin/xpack/application/analytics/post_event";
+
+    private PostAnalyticsEventAction() {
+        super(NAME, Response::new);
+    }
+
+    public static class Request extends ActionRequest {
+
+        private final String collectionName;
+
+        public Request(String collectionName) {
+            this.collectionName = collectionName;
+        }
+
+        public Request(StreamInput in) throws IOException {
+            super(in);
+            this.collectionName = in.readString();
+        }
+
+        public String collectionName() {
+            return collectionName;
+        }
+
+        @Override
+        public ActionRequestValidationException validate() {
+            return null;
```

Comment:
Should we validate that `collectionName` is non-empty, does not have any disallowed characters?